discovery.md

```diff
@@ -101,18 +101,295 @@ be considered an event subscriber and a target user of this API.
+
+#### `/providers/{name}`
+
+Retrieves an individual provider entity by exact match on the `provider` name.
+
+##### Returns
+
+A `provider` entity as a JSON object.
+
+#### `/providers/{name}/types`
+
+Retrieves the details about the types that are offered by the specified
+`provider`.
+
+##### Returns
+
+A JSON array of `type` entities
+
+#### `/producer/{provider.name}/{type.name}`
+
+Retrieves the `producer` entity that specifies the information necessary to
+create subscriptions and to consume the events. The `provider.name` and
+`type.name` items in the request path correspond to the information that can
+be obtained via the `/provider` and `/type` API calls.
+
+##### Returns
+
+A `producer` entity as a JSON object.
+
+### Entity Specifications
+
+This section details the fields that make up each of the entities referenced
+earlier in this document.
+
+#### `provider` entity
+
+Used in discovery for enumerating the different providers represented in this
+discovery system.
+
+##### `provider` entity attributes
+
+###### provider
+
+- Type: `String`
+- Description: Identifies the provider of the event by nane. The producer string
+  SHOULD be human readable and can identify a top level producer system.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string
+- Examples:
+  - "GitHub.com"
+  - "Awesome Cloud Storage"
+  - "com.example.microservices.userlogin"
+
+###### description
+
+- Type: `String`
+- Description: Human readable description .
+- Constraints:
+  - OPTIONAL
+  - If present, MUST be a non-empty string
+
+###### provider_uri
+
+- Type: `URI`
+- Description: Absolute URI that provides a link back to the producer, or
+  documentation about the producer. This is for human consumption.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST be a non-empty URI
+- Examples:
+  - "http://cloud.example.com/docs/blobstorage"
+
+
+##### `provider` entity examples
+
+```json
+{
+  "provider": "Cloud Storage Provider",
+  "description": "Blob storage in the cloud",
+  "provider_uri": "https://cloud.example.com/docs/storage"
+}
+```
+
+And a list of valid `provider` entities.
+
+```json
+[
+  {
+    "provider": "Cloud Storage Provider",
+    "description": "Blob storage in the cloud",
+    "provider_uri": "https://cloud.example.com/docs/storage"
+  },
+  {
+    "provider": "Cloud MySQL"
+  },
+  {
+    "provider": "Cloud OtherSQL",
+    "description": "Highly scalable SQL service"
+  }
+]
+```
+
+#### `type` entity
+
+Used in discovery for enumerating the different CloudEvent `type`s represented
+in this discovery system.
+
+##### `type` entity attributes
+
+###### type
+
+- Type: `String`
+- Description: CloudEvents [`type`](https://github.com/cloudevents/spec/blob/master/spec.md#type)
+  attribute.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string, following the constraints as defined in the
+    CloudEvents spec.
+- Examples:
+  - "com.github.pull.create"
+  - "com.example.object.delete.v2"
+
+###### specversion
```

Comment:
Need to think about this more, but it kind of feels like specversion should be more of a global kind of thing and not something that would be Type specific - or at least linked more with the Provider than with the type of the event.

Reply:
makes sense - dropped it from here. `specversion` is still available in the producer entity